db/migrate/20231210061330_add_created_at_to_fts_targets_with_index.rb

```diff
@@ -0,0 +1,106 @@
+# For auto load
+FullTextSearch::Migration
+
+class AddCreatedAtToFtsTargetsWithIndex < ActiveRecord::Migration[5.2]
+  def up
+    return if !table_exists?(:fts_targets)
+
+    add_column :fts_targets, :created_at, :timestamp
+
+    ActiveRecord::Base.transaction do
+      update_created_at_using_last_modified_at
+      update_created_at_using_created_on
+    end
+
+    if Redmine::Database.mysql?
+      add_index :fts_targets, :created_at
+    else
+      remove_index :fts_targets, name: "fts_targets_index_pgroonga"
+      add_index :fts_targets,
+                [:id,
+                 :source_id,
+                 :source_type_id,
+                 :project_id,
+                 :container_id,
+                 :container_type_id,
+                 :custom_field_id,
+                 :is_private,
+                 :last_modified_at,
+                 :created_at,
+                 :title,
+                 :content,
+                 :tag_ids],
+                using: "PGroonga",
+                with: "normalizer = 'NormalizerNFKC121'",
+                name: "fts_targets_index_pgroonga"
+    end
+  end
+
+  def down
+    return if !table_exists?(:fts_targets)
+
+    if Redmine::Database.mysql?
+      remove_index :fts_targets, :created_at
+    else
+      remove_index :fts_targets, name: "fts_targets_index_pgroonga"
+      add_index :fts_targets,
+                [:id,
+                 :source_id,
+                 :source_type_id,
+                 :project_id,
+                 :container_id,
+                 :container_type_id,
+                 :custom_field_id,
+                 :is_private,
+                 :last_modified_at,
+                 :title,
+                 :content,
+                 :tag_ids],
+                using: "PGroonga",
+                with: "normalizer = 'NormalizerNFKC121'",
+                name: "fts_targets_index_pgroonga"
+    end
+
+    remove_column :fts_targets, :created_at
+  end
+
+  private
+
+  class TmpTarget < ActiveRecord::Base
+    self.table_name = 'fts_targets'
+  end
+
+  class TmpType < ActiveRecord::Base
+    self.table_name = 'fts_types'
+  end
+
+  def update_created_at_using_last_modified_at
+    types = ['Attachment', 'Change', 'Changeset', 'CustomValue', 'Document', 'Journal', 'News']
+
+    TmpType.where(name: types).pluck(:id).each do |type_id|
+      TmpTarget.where(source_type_id: type_id).in_batches do |targets|
+        targets.update_all('created_at = last_modified_at')
+      end
+    end
```

Comment:
Does `TmpTarget.where(source_type_id: TmpType.where(name: types).select(:id)).update_all(created_at: TmpTarget.arel_attribute("last_modified_at"))` work?

Reply:
fix: [00926ae](https://github.com/clear-code/redmine_full_text_search/pull/118/commits/00926ae95772cb34e1f4a59f22d91bfb49ebcb89)
Thank you so much for giving me the new option.
The `arel_attribute` will be deprecated in Rails version 7 so I used the arel_table instead of it.
- ref: https://github.com/rails/rails/pull/39881